Automattic/wp-calypso
# Image editor: Uploading and rotating/flipping images from the local file system

Issue #2098

* Hidden behind "image-editor" flag
* Adds the image editor view, containing rotate and flip options
![3](https://cloud.githubusercontent.com/assets/800604/15026147/de93710c-1234-11e6-8eaa-bdfb6aa00fb9.png)
* Adds "Edit image" button to the image details modal - saving doesn't work at the moment due to the cross origin requests
![1](https://cloud.githubusercontent.com/assets/800604/15026143/d6b7e468-1234-11e6-8fbc-87e678057189.png)
* Adds "Add and edit image" to the top of media gallery - at the moment this is just for presentation purposes and allows to drag & drop image files into the editor. Images edited using this method can be uploaded to the server
![2](https://cloud.githubusercontent.com/assets/800604/15026146/db3124f0-1234-11e6-9219-5954adf1c8ff.png)

## Testing Instructions
- Start Calypso with `ENABLE_FEATURES=post-editor/image-editor make run`
- Navigate to calypso.localhost:3000/post
- Select a site when prompted
- Click on the media gallery button in the editor
- Click on "Add and edit image" (temporary workflow)
- Drag and Drop an image
- Rotate/Flip
- Verify that the modified image was uploaded
